### PR TITLE
fix(cts): improve oneOf matching with enum

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1384](https://algolia.atlassian.net/browse/DI-1384)

Fix oneOf matching for `EventItem`, by checking that the enum string is allowed.
It uses the tests cases added and removed in #1647.

Also clean up unused lambdas.

[DI-1384]: https://algolia.atlassian.net/browse/DI-1384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java [hunk 1]
```diff
@@ -85,9 +86,6 @@ protected Builder<String, Lambda> addMustacheLambdas() {
     lambdas.put("escapeQuotes", new EscapeQuotesLambda());
     lambdas.put("escapeSlash", new EscapeSlashLambda());
     lambdas.put("replaceBacktick", new ReplaceBacktickLambda());
-    lambdas.put("escapeReserved", new EscapeDartLambda());
-    lambdas.put("escapeDollarSign", new EscapeDollarSignLambda());
-    lambdas.put("lowerCamelCase", new LowerCamelCaseLambda());
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java:90]: do you know the use case for those lambdas @aallam ?

**Review by shortcuts — COMMENTED** [review]
INSANE

there's so many cases where this will allow us to write more tests

**Comment by millotp** [comment]
yes this is like a discriminator, but automatic 😀

File: generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java (hunk 1 above)
Comment (aallam) [reply generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java:90]: I remember some of them where used for dart, but I can't find the usage anymore! maybe got refactored away!

**Review by shortcuts — APPROVED** [review]
GGG!!

(just realized PHP tests are not formatted, that's weird)

**Comment by millotp** [comment]
yeah I'm not sure why I'll try to make it work in another PR

**Pull request merged**